.github/workflows/cypress-test-oidc-e2e.yml

```diff
@@ -200,7 +200,7 @@ jobs:
       - name: Configure and Run OpenSearch Dashboards with Cypress Test Cases
```

Comment:
I'm seeing a lot of redundancy exposed in this change.  Seems like if we compressed all these cypress tests into one file we could reuse some workflow jobs.

In the backend part of the project, we've got a single workflow step [1] that figures out what the test jobs should be run, [2] the execution of tests on all platforms, and [3] collecting all the test results to publish code coverage.  There are similar patterns with all these cypress tests

- [1] https://github.com/opensearch-project/security/blob/b8a315a321927a8f4490fb5a7023663ffc184e31/.github/workflows/ci.yml#L16
- [2] https://github.com/opensearch-project/security/blob/b8a315a321927a8f4490fb5a7023663ffc184e31/.github/workflows/ci.yml#L35
- [3] https://github.com/opensearch-project/security/blob/b8a315a321927a8f4490fb5a7023663ffc184e31/.github/workflows/ci.yml#L70

Let me know what you think.

Reply:
Moved it into a separate workflow. Thus, other tests can even rely on this, or even other plugins who want to do a similar check. If that ends up being the case, we can look into investing time into making it more extensible, but I think the PR is in a good state to get merged now, thanks!